src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,103 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (Movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setHaveTitle] = useState('');
+  const [description, setHaveDescription] = useState('');
+  const [imageUrl, setHaveImageUrl] = useState('');
+  const [imdbUrl, setHaveImdbUrl] = useState('');
+  const [imdbId, setHaveImdbId] = useState('');
```

Comment:
It's better to use a single state object for all values, as it simplifies state management and reduces code complexity. It also minimizes unnecessary re-renders, as updating multiple values in one object triggers only one re-render. Additionally, you can create a variable for the initial state value, making the code more maintainable and reusable:

```suggestion
const initialFormData = {
  description: '',
  title: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: ''
};

const [formData, setFormData] = useState(initialFormData);
```